src/lib/mongoBackend/TriggeredSubscription.cpp

```diff
@@ -132,12 +134,7 @@ std::string TriggeredSubscription::toString(const std::string& delimiter)
 *
 * TriggeredSubscription::stringFilterSet - 
 */
-void TriggeredSubscription::stringFilterSet(StringFilter* stringFilterP)
+void TriggeredSubscription::stringFilterSet(StringFilter* _stringFilterP)
```

Comment:
At the end (dicussed at skype) we need to get back to clone semantics (cache refresh logic could remove the StringFilter while TriggeredSubscription is using it... and we already discarded the usage of sems for this while we worked on PR #1900).

Thus, explicit copy each field would be a more clear solution than the "obscure" `stringFilter = *stringFilterP`. However, it is not clear yet how to clone a `regex_t` (open question at SOF: http://stackoverflow.com/questions/36846426/best-way-of-cloning-compiled-regex-t-struct-in-c). If we don't find a better solution, then we need to re-compile it using regcomp() from the original string.